### PR TITLE
RA campaign - replace TargetAndAttack with a global aircraft attack function

### LINKED ISSUE #18261: Aircraft may sit idle in RA missions
Mentioned in https://github.com/OpenRA/OpenRA/issues/4989#issuecomment-633302909

Aircraft in RA missions may exhibit this behavior
- Pick a random target
- Attack the target
- Go re-arm, but if our target is still alive, sit on the airfield indefinitely (or until we're kicked off, then circle indefinitely).

Several RA missions use a [`TargetAndAttack` function](https://github.com/OpenRA/OpenRA/search?l=Lua&q=TargetAndAttack) for aircraft logic. As a proof of concept, this snippet below could be used in-place of the existing function to fix the problem. The only difference is the target will not persist after re-arming, and a new target is chosen (would be simple to support this).

```Lua
HuntWithAircraft = function(aircraft)
	Trigger.OnIdle(aircraft, function()
		Media.DisplayMessage(tostring(aircraft) .. ' idle triggered')

		local target = nil
		local enemies = Utils.Where(Map.ActorsInWorld, function(self) 
			return self.Owner == player and aircraft.CanTarget(self) 
		end)
		if #enemies > 0 then
			target = Utils.Random(enemies)
		end

		if target ~= nil then
			Media.DisplayMessage(tostring(aircraft) .. ' attacking target ' .. tostring(target))
			aircraft.Attack(target)
		else
			Media.DisplayMessage(tostring(aircraft) .. ' no targets found, returning to base.')
			aircraft.ReturnToBase()
		end	
	end)
end
```

I'd like to test this a bit longer, then submit a pull request to move this into a `campaign-global.lua` like we've done for TD and D2K.

If it's worth doing, aircraft in the original campaign also seem to prioritize certain actors (e.g. defenses in soviet09), and we could expand this to support it.

Interested in feedback from any mission maintainers.

### PR DESCRIPTION
**Pull request by cjshmyr**
Fixes #18261.

This replaces `TargetAndAttack` with `InitializeAttackAircraft`. The old function is currently broken due to an aircraft activity overhaul; the new function sets up an idle trigger to follow the same behavior as the old function.

`InitializeAttackAircraft` has been added to a new `campaign-global.lua` file. It also ignores wall targets which was a feature of Allies08a/b. Ignoring walls could be designed better than how I did it.

I tested each mission touched, and also attached a test case commit that demonstrates this behavior (will remove it later). 

Note that as I make this PR, allies05 is not testable as-is since harvester search range is busted for several campaign maps. This should be handled by a separate PR.

### CODE REVIEW

File: mods/ra/bits/scripts/campaign-global.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ChooseRandomTarget = function(unit, enemyPlayer)
+	local target = nil
+	local enemies = Utils.Where(enemyPlayer.GetActors(), function(self)
+		return self.HasProperty("Health") and unit.CanTarget(self) and not Utils.Any({"sbag","fenc","brik","cycl","barb"}, function(type) return self.Type == type end)
+	end)
+	if #enemies > 0 then
+		target = Utils.Random(enemies)
+	end
+	return target
+end
```
Comment (Smittytron) [inline mods/ra/bits/scripts/campaign-global.lua]: newline

**Review by Smittytron — DISMISSED** [review]
This works as advertised. Will be a huge step towards getting several of these missions out!

**Comment by cjshmyr** [comment]
Updated and test case removed

**Review by abcdefg30 — APPROVED** [review]
Lgtm.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1baa9873d686d79188f4fa2e032fd3fa3610f161...4a8370670b4bc51d9711d40bde9dc91f0242559f)

**Pull request merged**